### ISSUE TITLE
Can't reach the first key in outline

### ISSUE THREAD

**Issue by dangh** [zed-industries/community#1771]
### Check for existing issues

- [X] Completed

### Describe the feature

I can't reach the first "type" key in this JSON document with outline:

```json
{
	"type": "string",
	"at": "some where",
	"nested": {
		"type": "number"
	}
}
```

- Type `t`, the cursor in the buffer jump to "at".
- Type `ype`, zed focus the second "type".

I thought Zed does an incremental search so it jump to second "type" after focusing "at", but even typing `ype` from the beginning will jump to second "type".

**Comment by maxbrunsfeld** [comment]
You can still select it with the arrows, right? The problem is just that the first one is selected by default?

**Comment by dangh** [comment]
@maxbrunsfeld Yes, correct. After typing the keyword to filter the entries, I have to use the arrow key to pick the first one. It's quite frustrating for long documents.
